pkg/storageincentives/agent.go

```diff
@@ -400,6 +398,8 @@ func (a *Agent) handleSample(ctx context.Context, round uint64) (bool, error) {
 		return false, nil
 	}
 
+	a.state.SetLastSelectedRound(round + 1)
```

Comment:
I agree with the +1, but why move it from the start until after the validation?   IMHO the node/neighborhood is selected even if it is not fully synced.  If someone is watching this value to determine the frequency of neighborhood selection, it is no longer correct.

On the other hand, if we are not fully synced and therefore have the wrong idea of the storage radius, we may think we are selected when in fact we are not.   I can see it both ways.